app/models/loader/types.rb

```diff
@@ -199,8 +199,18 @@ def create!
     class User < Record
       def_delegators :@policy_object, :public_keys, :account, :role_kind, :uidnumber, :restricted_to
 
+      def check_user_creation_allowed(res_id:)
+        user_creation_allowed = ENV.fetch('CONJUR_ALLOW_USER_CREATION', 'true').downcase
+        if user_creation_allowed == 'false' && res_id.include?('@')  # not under root
```

Comment:
Although not conventional, I really don't agree with the change to prevent users from being created outside of the `root` policy.  As this breaks existing functionality, if we're going to do this, we need to do it in a very visible way and with alignment from PO/PM. Customer need to be aware it could break their existing workflows.

I know that users need to be part of the root policy in order to work with Idaptive. If we also want to limit created users to the `root` policy, what do you think about extending our settings to prevent creation of users outside of the `root` policy?

Reply:
Upon re-reading this code, I realize I misread the intent of this block.  Instead of including the check for a user in a non-root policy, how about returning, something like:

```ruby
def check_user_creation_allowed(resource_id:)
  return if resource_id.include?('@')

  if ENV.fetch('CONJUR_ALLOW_USER_CREATION', 'true').downcase == 'false'
    raise Exceptions::InvalidPolicyObject.new(
      res_id,
      message: 'User creation through policy is disabled.'
    )
  end
end
```
Personally, it helps me to see where the exits are without needing to think through the `&&` portion of the `if` statement.  it also highlights that we need separate unit tests for the possible results of `check_user_creation_allowed`:

1. User is outside the root policy => do nothing
2. User is inside the root policy and `CONJUR_ALLOW_USER_CREATION` is not set => do nothing
3. User is inside the root policy and `CONJUR_ALLOW_USER_CREATION` is not set to `true` => do nothing
4. User is inside the root policy and `CONJUR_ALLOW_USER_CREATION` is not set to false => throw exception

Reply:
Hi Jason
The required behavior is different then you describe
It is:
1. User is inside the root policy => do nothing
2. User is outside the root policy and CONJUR_ALLOW_USER_CREATION is not set => do nothing
3. User is outside the root policy and CONJUR_ALLOW_USER_CREATION is not set to true => do nothing
4. User is outside the root policy and CONJUR_ALLOW_USER_CREATION is not set to false => throw exception

Yet - I can align my code to the format you suggested

Reply:
Regarding the influence of this change on conjur-oss behaviour - there should not be any influence there (you thought it will be because your description of login did not mach the actual implementation